lib/bundler/vendor/compact_index_client/lib/compact_index_client/updater.rb

```diff
@@ -24,33 +26,41 @@ def initialize(fetcher)
     def update(local_path, remote_path, retrying = nil)
       headers = {}
 
-      if local_path.file?
-        headers["If-None-Match"] = etag_for(local_path)
-        headers["Range"] = "bytes=#{local_path.size}-"
-      else
-        # Fastly ignores Range when Accept-Encoding: gzip is set
-        headers["Accept-Encoding"] = "gzip"
-      end
+      Dir.mktmpdir(local_path.basename.to_s, local_path.dirname) do |local_temp_dir|
+        local_temp_path = Pathname.new(local_temp_dir).join(local_path.basename)
 
-      response = @fetcher.call(remote_path, headers)
-      return if response.is_a?(Net::HTTPNotModified)
+        # download new file if retrying
+        if retrying.nil? && local_path.file?
+          FileUtils.cp local_path, local_temp_path
```

Comment:
It ensures the view of `local_path` is consistent through the whole process - if the etag and .size were taken of the original local_path, then it might change during execution with another parallel process.

It also ensures that when we append the response to the file that it's for the same matching content. If it had changed in the meantime then it'd be corrupt.